aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsEndpointDecorator.kt

```diff
@@ -45,13 +47,15 @@ class AwsEndpointDecorator : RustCodegenDecorator {
     }
 
     override fun configCustomizations(
+        rustSettings: RustSettings,
```

Comment:
hmm, can we push `RustSettings` into `ProtocolConfig`?

Reply:
the other option could be to add an `initialize` method to the decorator so that `rustSettings` would be accessible as a regular member.

Reply:
Decided to revert the RustSettings changes for now. I've documented the approach we should take for this in #616.